tests/integration/select2-methods.js

```diff
@@ -137,3 +137,60 @@ test('multiple value matches the jquery value', function (assert) {
     'The values should match the jquery values'
   );
 });
+
+test('selection and clearing of data from ajax source', function (assert) {
```

Comment:
To prove that this branch fixes #5391, a second response from the AJAX API should be mocked-out -- with updated, differing metadata for one of the returned items -- and an assertion should be added to check that selecting the updated item is reflected on the control.